crates/wg_network/src/routing.rs

```diff
@@ -2,7 +2,7 @@ pub type NodeId = u8;
 
 #[derive(Debug, Clone)]
 pub struct SourceRoutingHeader {
-    pub hop_index: usize, // must be set to 0 initially by the sender
+    pub hop_index: usize, // must be set to 1 initially by the sender
```

Comment:
This is not true because of the drone rule

Reply:
I did not understand what you are referring to. Why shouldn't that be true because of the drone rules?

Reply:
Because the drone increase by one the index as first rules.

Reply:
No because the array is 0 indexed, so if we take your patch, on the first hop the index would be increase to 2 and then the next we use to decide to do forwarding is 3 (which is wrong).

If you also change the rules that will be fine, but maybe that could be better in another PR.

Reply:
I’m closing the conversation because I believe it was based on a partial reading of the changes. Feel free to reopen it if I misunderstood something.